core/src/main/java/hudson/PluginManager.java

```diff
@@ -207,6 +214,24 @@ public PluginManager(ServletContext context, File rootDir) {
         }
     }
 
+    /**
+     * Get the handler for UI layer
+     * @return Proxy for UI if not null, else this
+     */
+    @Override
+    public Object getTarget() {
+        return (proxy != null) ? proxy : this;
+    }
+
+    /**
+     * Set the proxy for plugin manager UI handling
+     * @param proxy UI handler, else null to remove the proxy
+     */
+    @Nullable
+    public void setTarget(Object proxy) {
```

Comment:
Not entirely sure this should be directly exposed - rather have another method that calls this that is then more descriptive and can be typed.